### DISCUSSION TITLE
Squash and Merge on Github Mobile doesn't match the same on web

### DISCUSSION THREAD

**Discussion by nicksloan** [community/community#51016] [Mobile]
### Select Topic Area

Product Feedback

### Body

When I Squash & Merged a PR on Github mobile, I expected the same experience as web: a prompt to edit/confirm the squashed commit message. I wasn't given a prompt to do so, but even worse: the squashed commit message does not match the default behavior of Github on the web. All of the squashed commit messages (which we use to close issues) were omitted rather than being included in the commit message. There was no way to know that this would happen, and no good way to recover from the broken behavior (rewriting the history on our protected main branch is not an acceptable recovery path for this issue).

Unfortunately I have to give my team guidance to **never** merge commits from Github Mobile. This feature is unusable in its current state.

**Comment by iskae** [comment]
Hello @nicksloan 👋 

Thanks so much for reaching out ❤️ 

>  I expected the same experience as web: a prompt to edit/confirm the squashed commit message

I believe you are right 👍 Currently we are not offering the users to change the commit message as part of `Squash and Merge` flow. It is a bit hidden when the user taps on the merge options button beside the `Squash and Merge` button as follows: 

https://user-images.githubusercontent.com/26570736/229737023-a89aead9-545c-4237-bcbe-bb19e6670c9f.mp4


> the squashed commit message does not match the default behavior of Github on the web. All of the squashed commit messages (which we use to close issues) were omitted rather than being included in the commit message.

Unfortunately I was not able to reproduce this behaviour. Can you please elaborate a bit more on this? 😊 

In any case I have created an internal issue for improving the user flow for merging on mobile 🙇 Please stay tuned for our future updates 😊 

Cheers!

**Reply by nicksloan** [reply]
I should have been more clear. By mobile, I mean the iOS app.

I've looked again, and I see that the option to edit the message does exist now behind a cog icon.

That said, I can't explain why the commit message when I squash and merged a commit previously doesn't seem to have matched the default format displayed in the view behind that icon.

**Reply by iskae** [reply]
> That said, I can't explain why the commit message when I squash and merged a commit previously doesn't seem to have matched the default format displayed in the view behind that icon.

🤔 That is unfortunate. Let's see if we can come up with a better flow for our users 🙇 

Thanks once again reaching out and please do not hesitate to do so in the future so that we can make our apps better ❤️

**Reply by MultiUniverse1974** [reply]
Delete Branch is the key

**Comment by MultiUniverse1974** [comment]
Delete Branch after what you just did on the video

**Reply by stevenh** [reply]
This just caught me out and as the original reporter said no way back so it classify it as quite a serious issue.

**Reply by stevenh** [reply]
I can also confirm the original posters issue with it not following the configured settings for what the default squash merge message is.

In my case it just contained the PR title and no other content, when it should have contained all commit messages concatenated, as per the repository settings.

**Reply by Schirvani62** [reply]
999
8998щ8щщщщззззззщ8зщщщзззз8зззщщз899000099099990з0зжз00зззз8х89жзжжжжш7ш щшщ
Щш
Шщжжжщщщ

Южшж
Щшшжзшжжзшжщжщщшщзжшщшжшщжжэжэжщжжжюжшжэзжщ щит99
Дзж9@

**Reply by Schirvani62** [reply]
Ншд

9

Дю6л97жо
жжш⅔[]()

**Reply by Schirvani62** [reply]
7098

Г99999
#  Здравствуйте 0эы

**Comment by jpeach** [comment]
> All of the squashed commit messages (which we use to close issues) were omitted rather than being included in the commit message

This reproduces 100% for me. 

Here's a trivial [PR](https://github.com/apache/trafficserver/pull/9849), note that the [commit](https://github.com/apache/trafficserver/pull/9849/commits/08b9f8dea820aa79eda4557799273a18508dbe90) has a message. I merged it on GitHub mobile - there was no prompt that showed the commit message and I got [this](https://github.com/apache/trafficserver/commit/6411b2603e82411bdf820934177d6349e7a00ca4) result. Note, that the content of the commit message it missing.

**Comment by rebello95** [comment]
I just experienced this exact issue on connectrpc/connect-swift#182. I verified in the popup demonstrated above that the commit message was set to the original PR description (which is the default rule on the repository), but when I actually tapped the "Squash and merge" button, the commit message was dropped from the commit on main entirely.

I've seen this on other commits in the Connect-Swift repository as well, seems pretty consistent.

**Comment by richardallred** [comment]
@iskae any chance this can be addressed soon? is there a way to create an actual Bug somewhere? This makes merging PRs via mobile essentially broken for my team's workflow as we rely on the content of the PR description getting into the commit message on merge.

**Comment by iskae** [comment]
Hi folks 👋 

We already have an internal issue for this and let me bump up the priority for it 🙇 I hope we get the chance to work on this soon and unblock everyone 😊 

Please stay tuned 🙇

**Comment by iskae** [comment]
Hey folks 👋 

We have been testing this on our end and we would love to dig this further to understand the exact problem. Here is a recording of the `Squash and Merge` flow on an Android emulator:

https://github.com/community/community/assets/26570736/05aab84c-59c5-498b-b530-a1832825ed99

During our tests the PR body was also included in the commit message. Would it be possible to let us know the exact steps that you are taking @richardallred ? Also are you using an Android or an iOS device?

Thanks a bunch in advance 🙇

**Reply by richardallred** [reply]
I'm on iOS, I can try to make a similar recording of the bug there later today.

**Comment by keerthanam12** [comment]
I'm Keerthana

**Comment by chubei-urus** [comment]
We can still reliably reproduce it on GitHub Mobile iOS. See the attached video. The commit message seems correct before merging, but is still empty after being merged, which is even worse than being empty in the beginning.

Note that the repository has a ruleset that applies to the default branch that only allows Squash merge.

https://github.com/user-attachments/assets/68fa37c7-2518-4bd7-906b-53519e074da1

**Comment by iskae** [comment]
Hi @richardallred @chubei-urus 👋 

We are aware of these issues and working on a fix 😊 Hopefully the fix will be available in the upcoming weeks 🎉 

Please stay tuned and sorry for causing trouble for you folks ❤️

**Reply by binhonglee** [reply]
just noticed this issue (especially troublesome when used with entire.io). it's been over a year now, any progress?

**Comment by Shinrai** [comment]
4 year old issue and still coming back... had several squash merges which worked just fine then all the sudden all of them this month are merging with no body... 

![image](https://github.com/user-attachments/assets/ded38587-533f-4b0e-98e3-ed0cd2fe5559)

![image](https://github.com/user-attachments/assets/7f3f626e-9a1a-4485-ab3b-dd6f1eaaeb19)

![image](https://github.com/user-attachments/assets/4a321f0b-22f2-4a2b-87ff-18423d031414)
